From b4f95594634eb38421d0a7e8228b1ea43841ff68 Mon Sep 17 00:00:00 2001
From: ofer benjamin <obenjamin@habana.ai>
Date: Mon, 29 Jun 2020 12:08:00 +0300
Subject: [PATCH] [SW-0] dsv: add dsv.mk

Add dsv.mk to generate the dsv-fit.itb file. The file will be creat inside
the dsv_custom folder and copy to the output/images folder

Change-Id: I3358d13e9bced31fcb1f567db4151dc9ab526a19
Signed-off-by: ofer benjamin <obenjamin@habana.ai>
---

diff --git a/package/dsv/dsv.mk b/package/dsv/dsv.mk
index d46bfb3..3c62b15 100644
--- a/package/dsv/dsv.mk
+++ b/package/dsv/dsv.mk
@@ -1,9 +1,26 @@
 DSV_VERSION = 1.0.0
 DSV_SITE = dsv
 DSV_LICENSE = unclear
+DSV_ITB_FILE := dsv-fit.itb
+
+COMMON_BASE := $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common
+DSV_ITS_FILE := dsv.its
+DSV_ITS_TEMPLATE := $(COMMON_BASE)/$(DSV_ITS_FILE)
+
+DSV_DEPENDENCIES = host-dtc host-uboot-tools
+
+DSV_INSTALL_IMAGES = YES
+
+define DSV_INSTALL_IMAGES_CMDS
+	$(INSTALL) -D -m 0644 $(@D)/$(DSV_ITB_FILE) $(BINARIES_DIR)/$(DSV_ITB_FILE)
+endef
 
 define DSV_BUILD_CMDS
-	$(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D) all
+	($(TARGET_MAKE_ENV) $(MAKE) CROSS_COMPILE="$(TARGET_CROSS)" -C $(@D) all; \
+		cp -p $(DSV_ITS_TEMPLATE) $(@D)/ ; \
+		cd $(@D); \
+		$(HOST_MAKE_ENV) $(HOST_DIR)/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f $(DSV_ITS_FILE) $(DSV_ITB_FILE); \
+	)
 endef
 
 $(eval $(generic-package))
